### PR TITLE
initial development

### PR DESCRIPTION
**Pull request by jbrestel**

### CODE REVIEW

File: bin/calcStats [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+open (OUT, ">>$outFile") or die "Can't open '$outFile' for writing\n";
+
+##########################
+##### "main()" ###########
+
+if ( -f $dataset ) {
+    warn "Processing $dataset\n" if DEBUG;
+    processOneFile($dataset, \*OUT);
+} elsif ( -d $dataset ) {
```
Comment (jbrestel) [inline bin/calcStats]: i think the input for this script is one file.  why the directory processing?

File: bin/calcStats [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    my ($inputFile, $fh) = @_;
+
+    $inputFile =~ s/(.*\.gz)\s*$/gzip -dc < $1|/;
+    $inputFile =~ s/(.*\.Z)\s*$/uncompress -c < $1|/;
+    
+    my $seqIO = Bio::SeqIO->new(-file => $inputFile);
+    my $pIcalc = Bio::Tools::pICalculator->new();
+
+    while (my $richSeqObj = $seqIO->next_seq) {
+	my $seq = Bio::Seq->new(-id => $richSeqObj->id,
```
Comment (jbrestel) [inline bin/calcStats:46]: The seqio->nextseq gives you a Bio::Seq object.  you should be able to just set the alphabet there (if it isn't already set.  You don' t need to make another Bio::Seq object

File: main.nf [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  throw new Exception("Missing params.inputFilePath")
+}
+
+//--------------------------------------------------------------------------
+// Main Workflow
+//--------------------------------------------------------------------------
+
+workflow {
+  tabfiles = findStats(seqs)
+  catFiles(tabfiles.collectFile(), params.outputFileName)
```
Comment (jbrestel) [inline main.nf]: remove this as "collectFile" does the concatenation.  
collectFile(name: params.outputFileName, storeDir: params.outputDirectory)

File: nextflow.config [hunk 4]
```diff
@@ -0,0 +1,9 @@
+params {
+  inputFilePath = "$baseDir/data/input.fa"
+  fastaSubsetSize = 1
+  outputFileName = "proteinStats.tab"
+  outputDir = "$launchDir/output"
+}
+docker {
```
Comment (jbrestel) [inline nextflow.config]: replace this with a call to include the conf/docker.config

File: bin/calcStats (hunk 1 above)
Comment (steve-fischer-200) [reply bin/calcStats]: copied from orfFinder

File: main.nf (hunk 3 above)
Comment (steve-fischer-200) [reply main.nf]: how does it know to collect tabfiles?

File: nextflow.config (hunk 4 above)
Comment (steve-fischer-200) [reply nextflow.config]: you'll have to explain this.

File: main.nf (hunk 3 above)
Comment (jbrestel) [reply main.nf]: it uses "cat" i believe

File: nextflow.config (hunk 4 above)
Comment (jbrestel) [reply nextflow.config]: replace the docker {} bit with:

includeConfig "$baseDir/conf/docker.config"

NOTE: this is also wrong in orffinder.  i'll change that one too

File: bin/calcStats (hunk 1 above)
Comment (jbrestel) [reply bin/calcStats]: i just updated orfFinder to only take an "inputFile".  There is no need to have the directory processing there.  i had just moved over the existing script but better to remove it

**Pull request merged**